content/francineblanc/how-to-use-git-for-beginners.md

```diff
@@ -0,0 +1,219 @@
+
+### Merging
+Merging combines multiple commits into one history, and is generally used to join or combine two branches. So, when ready for work on a branch to be merged into ```main```, the command ```git merge branch-name``` is used, where ```branch-name``` is the name of the branch to be merged into the ```main``` branch. This command is most often used when working on a project with others (more on this below).
+
+Once a branch has been merged, it is no longer needed so can be deleted with:
+
+```
+git branch -d <branch-name>
+``` 
+This will though only delete the branch once it has been merged into ```main```. Replacing the ```-d``` flag with ```-D``` will delete the branch even if it hasn't been merged with ```main```.
+
+## Collaboration
+When collaborating with others on a project, it is generally the case that a shared remote Git repository will exist so that multiple people can work on the same project from different locations. There are a wide range of providers of remote repositories but the most popular and well-known is [GitHub].
+
+### Git != GitHub
+When starting out, it is easy to get confused between Git and GitHub but they are not the same. Git is a version control tool whereas GitHub is an onlne hosting service for Git repositories that has tools which integrate with Git. Git does not require GitHub to work and a remote repository is not needed for using Git however, when working with others on a project, GitHub (and any similar provider) makes doing so much easier.
+
+### Workflow
+When working with others, the general workflow is as follows:
+1. **Create a Remote Repository** - where a remote repo does not already exist, one is generally created by one of the people working on the project using a hosting provider;
+2. **Pull Changes to a Local Repo** - combines changes from the Remote Repo into a local branch
+3. **Copy/Clone the Remote Repository to a Local Repo** - this replicates and copies everything in the remote repository to a local copy of the Git project.
+4. **Create a new Local Branch** - a new branch will generally be made, forked from ```main```, and the new feature or fix will be worked on;
+5. **Add and Commit Changes** - the normal Git workflow is followed for adding and committing changes;
+6. **Push Changes to the Remote and Create a Pull Request** - the changes made on the local branch are pushed up to the remote branch to be reviewed by team members;
+7. **Merge the Local Branch with Main** - once the changes have been approved by team members, the local branch is merged into the main branch.
+
+Let's take a closer look at steps 2 and 5.
+
+### Fetching/Merging/Pulling
+To make sure the local repo is up to date with the remote, changes need to be fetched and merged from the remote and integrated into the local branch. This can be done in two stages using ```git fetch``` and ```git merge``` or in one step using ```git pull```.
```

Comment:
```suggestion
To keep the local repo up to date with the remote repo, changes can be fetched and merged from the remote and integrated into the local branch. This can be done in two stages using `git fetch` and `git merge` or in one step using `git pull`.
```